### DISCUSSION TITLE
Working with Vyper Version `3.1`

### DISCUSSION THREAD

**Discussion by defido** [pcaversaccio/xdeployer#37] [Show and tell]
```
const config: HardhatUserConfig = {
  solidity: {
    version: "0.8.9",
    settings: {
      optimizer: {
        enabled: true,
        runs: 200,
      },
    },
  },
  vyper: { <--- Adding in the vyper 
    version: "0.3.1",
  },
```
Add the vyper contract to your contracts folder. Example: `./contracts/Greeter.vy`

`https://www.npmjs.com/package/@nomiclabs/hardhat-vyper` <--- Add the hardhat Vyper compiler

`npx hardhat xdeploy` <--- Build & deploy

Forge also compiles & works. 

Verification is another story. :)

**Comment by defido** [comment]
Vyper is awesome.

**Reply by pcaversaccio** [reply]
![20220420_083343.jpg](https://user-images.githubusercontent.com/25297591/164165525-1a055fd2-ca77-4e71-b137-2a63c9595855.jpg)

**Reply by defido** [reply]
Step 1: Take the holiday
![image](https://user-images.githubusercontent.com/95591037/164166095-1a411004-c64f-408c-a4b4-916a0cd6b17c.png)
